### PR TITLE
feature: provide mock module

### PR DESCRIPTION
**Pull request by l81893521**
### Ⅰ. Describe what this PR did
Provide mock module for increase unit test coverage rate.

### Ⅱ. Does this pull request fix one issue?

#1436

### CODE REVIEW

File: rm-datasource/src/test/java/io/seata/rm/datasource/mock/MockDriver.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            LOGGER.error("please initialize the column meta and index meta");
+            throw new RuntimeException("please initialize the column meta and index meta");
+        }
+        MockResultSet resultSet = new MockResultSet(stmt, labels);
+
+        List<ResultSetMetaDataBase.ColumnMetaData> columns = null;
+        try {
+            columns = resultSet.getMockMetaData().getColumns();
+        } catch (SQLException e) {
+            LOGGER.error(e.getMessage());
```
Comment (xingfudeshi) [inline rm-datasource/src/test/java/io/seata/rm/datasource/mock/MockDriver.java]: use three parameter method,like LOGGER.error("XXXX:{}",e.getMessage(),e)

Comment (l81893521) [reply rm-datasource/src/test/java/io/seata/rm/datasource/mock/MockDriver.java]: Thanks for you advice.

File: rm-datasource/src/test/java/io/seata/rm/datasource/sql/struct/TableMetaTest.java [hunk 2]
```diff
@@ -318,6 +288,7 @@ public ResultSet executeQuery(MockStatementBase statement, String s) throws SQLE
                 new Object[] {"", "", "t1", "name3", Types.VARCHAR, "VARCHAR", 64, 0, 10, 0, "", "", 0, 0, 64, 5, "YES", "NO"},
                 new Object[] {"", "", "t1", "id_card", Types.DECIMAL, "DECIMAL", 20, 0, 10, 1, "", "", 0, 0, 64, 4, "NO", "YES"}
             };
+        mockDriver.setColumnMetas(columnMetas);
```
Comment (slievrly) [inline rm-datasource/src/test/java/io/seata/rm/datasource/sql/struct/TableMetaTest.java]: Here is an array of references, need to be set each time?

Comment (l81893521) [reply rm-datasource/src/test/java/io/seata/rm/datasource/sql/struct/TableMetaTest.java]: Yes, If you want to mock change the table struct

**Pull request merged**